v2/internal/controllers/crd_signalr_test.go

```diff
@@ -105,3 +129,68 @@ func Test_SignalRService_SignalR_CRUD(t *testing.T) {
 	tc.Expect(retryAfter).To(BeZero())
 	tc.Expect(exists).To(BeFalse())
 }
+
+func SignalR_SecretsWrittenToSameKubeSecret(tc *testcommon.KubePerTestContext, signalR *signalrservice.SignalR) {
+	old := signalR.DeepCopy()
+	signalrSecret := "signalrsecret"
+	signalR.Spec.OperatorSpec = &signalrservice.SignalROperatorSpec{
+		Secrets: &signalrservice.SignalROperatorSecrets{
+			PrimaryKey: &genruntime.SecretDestination{
+				Name: signalrSecret,
+				Key:  "primarykey",
+			},
+			PrimaryConnectionString: &genruntime.SecretDestination{
+				Name: signalrSecret,
+				Key:  "primaryconnectionstring",
+			},
+			SecondaryKey: &genruntime.SecretDestination{
+				Name: signalrSecret,
+				Key:  "secondarykey",
+			},
+			SecondaryConnectionString: &genruntime.SecretDestination{
+				Name: signalrSecret,
+				Key:  "secondaryconnectionstring",
+			},
+		},
+	}
+	tc.PatchResourceAndWait(old, signalR)
+
+	tc.ExpectSecretHasKeys(signalrSecret, "primarykey", "primaryconnectionstring", "secondarykey", "secondaryconnectionstring")
+}
+
+func SignalR_SecretsWrittenToDifferentKubeSecrets(tc *testcommon.KubePerTestContext, signalR *signalrservice.SignalR) {
+	old := signalR.DeepCopy()
+	primaryKeySecret := "secret1"
+	primaryConnectionString := "secret2"
+	secondaryKeySecret := "secret3"
+	secondaryConnectionString := "secret4"
+
+	// Not testing port as it's not returned by default so won't be written anyway
```

Comment:
I think delete this comment as it doesn't apply here?

Reply:
💯